js/common/lib/backend.ts

```diff
@@ -44,6 +68,11 @@ export interface Backend {
 
   createSessionHandler(uriOrBuffer: string|Uint8Array, options?: InferenceSession.SessionOptions):
       Promise<SessionHandler>;
+
+  createTrainingSessionHandler(
```

Comment:
should be optional `createTrainingSessionHandler?(...` instead of `createTrainingSessionHandler(...`

Reply:
this allows no change under /js/node and /js/react_native.

Reply:
This suggestion is much cleaner than what I had before -- pushed in the latest commit